src/main/java/appeng/blockentity/crafting/CraftingStorageBlockEntity.java

```diff
@@ -19,60 +19,11 @@
 package appeng.blockentity.crafting;
 
 import net.minecraft.core.BlockPos;
-import net.minecraft.world.item.Item;
 import net.minecraft.world.level.block.entity.BlockEntityType;
 import net.minecraft.world.level.block.state.BlockState;
 
-import appeng.block.crafting.AbstractCraftingUnitBlock;
-import appeng.core.definitions.AEBlocks;
-
 public class CraftingStorageBlockEntity extends CraftingBlockEntity {
```

Comment:
Is this class still necessary?

Reply:
This class is what CraftingService#updateCPUClusters looks for to retrieve valid crafting CPUs, so it's probably worth keeping around still for the sake of that distinction.

Reply:
I think it could be made to use `grid.getMachines(CraftingBlockEntity.class)` instead. From the implementation, this should also work for subclasses. Would also need to check that a cluster without storage doesn't assemble, but that should already be checked somewhere else.